pandas/_libs/parsers.pyx

```diff
@@ -2006,19 +2003,14 @@ def sanitize_objects(ndarray[object] values, set na_values,
         Py_ssize_t i, n
         object val, onan
         Py_ssize_t na_count = 0
-        dict memo = {}
 
     n = len(values)
     onan = np.nan
 
     for i in range(n):
         val = values[i]
-        if (convert_empty and val == '') or (val in na_values):
+        if val in na_values:
             values[i] = onan
             na_count += 1
-        elif val in memo:
```

Comment:
Anyone know what this code was supposed to do? Removing it doesn't break any tests and it looks like a no-op...

Reply:
If I understand it correctly, memo would ensure that duplicate complex objects (dicts/lists/... inside the ndarray) are all the same (instead of being duplicates that potentially point to different memory blocks).